packages/domain/lib/use_case/request_use_case_coming.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:domain/const/configuration.dart';
+import 'package:domain/entity/movie_coming_response.dart';
+import 'package:domain/repository/network_repository.dart';
+import 'package:domain/use_case/sample_use_case/use_case_out.dart';
+
+class RequestUseCaseComing
+    extends UseCaseOut<Future<List<MovieResponseComing>>> {
+  final NetworkRepository _networkRepository;
+
+  RequestUseCaseComing(
+    this._networkRepository,
+  );
+
+  @override
+  Future<List<MovieResponseComing>> call() async {
+    final response = await _networkRepository.requestMovie(
+      apiPath: Configuration.apiPathComingSoon,
+      queryParameters: Configuration.queryEmpty,
+    );
+    final List<MovieResponseComing> movieComing = [];
+    final countMovie = (response.headers['x-pagination-limit']).first;
```

Comment:
here according with task u should to read header 'X-Pagination-Limit' to know count of pages available with default items per page count. then if pages >= 5 make request with query parameter 'limit = 50'.
else u should read header 'X-Pagination-Item-Count' and make request with parameter limit = X-Pagination-Item-Count. If my remember do not trick me)